handlers.go

```diff
@@ -130,28 +130,31 @@ func (a *autographer) handleSignature(w http.ResponseWriter, r *http.Request) {
 			return
 		}
 
-		// Find the ID of the requested signer
-		// Return an error if the signer is not found or if the user is not allowed
-		// to use this signer
-		signerID, err := a.getSignerID(userid, sigreq.KeyID)
-		if err != nil || signerID < 0 {
+		// returns an error if the signer is not found or if
+		// the user is not allowed to use this signer
+		requestedSigner, err := a.authBackend.getSignerForUser(userid, sigreq.KeyID)
+		switch err {
+		case nil:
+			// got the signer and allowed to use it
+		default:
 			httpError(w, r, http.StatusUnauthorized, "%v", err)
 			return
 		}
```

Comment:
did you mean to use `if err != nil` ?

Reply:
Yep! Had the type switch since I was catching other types in development.